magenta/js/core/src/sequences.ts

```diff
@@ -157,11 +171,10 @@ export class Sequences {
    * @throws {NegativeTimeException} If a note or chord occurs at a negative
    *    time.
    */
-  public static quantizeNoteSequence(noteSequence: NoteSequence,
+  public static quantizeNoteSequence(noteSequence: INoteSequence,
     stepsPerQuarter: number): NoteSequence {
     // Make a copy.
-    const qns = NoteSequence.fromObject(
-      NoteSequence.toObject(noteSequence));
+    const qns = NoteSequence.fromObject(noteSequence);
```

Comment:
This actually isn't a deep copy, which is why I was using clone in MusicVAE.

Reply:
This ended up being a huge pain. Yes, it does turn out that fromObject doesn't do a deep copy unless you first do a toObject, but that's not necessarily available on an INoteSequence.

However, using the 'clone' package on an INoteSequence that is actually a NoteSequence object messes up the prototype somehow. I don't understand exactly what's happening, but the end result is that if your object had a repeated field that was empty before you cloned it, you will not be able to add to that repeated field after you clone it. It has something to do with how the clone package clones the prototype for the object.

Anyway, after trying a bunch of things, I'm now more or less convinced that the easiest/safest way to clone a proto object is to just encode/decode it. Another option would be something like:

```
clone(NoteSequence.toObject(NoteSequence.create(noteSequence)));
```

Or we could add special logic for different behavior depending on whether the object is actually a NoteSequence or just an object that matches INoteSequence.

But I think I prefer the encode/decode option because it works equally well with any kind of INoteSequence and is guaranteed to be a deep copy without any references to the original object. Thoughts?